impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/changeInterceptorBinding/ChangeInterceptorBindingTest.java

```diff
@@ -0,0 +1,36 @@
+package org.jboss.cdi.tck.tests.build.compatible.extensions.changeInterceptorBinding;
+
+import static org.testng.Assert.assertEquals;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.cdi.tck.AbstractTest;
+import org.jboss.cdi.tck.cdi.Sections;
+import org.jboss.cdi.tck.interceptors.InterceptorsSections;
+import org.jboss.cdi.tck.shrinkwrap.WebArchiveBuilder;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.test.audit.annotations.SpecAssertion;
+import org.jboss.test.audit.annotations.SpecVersion;
+import org.testng.annotations.Test;
+
+/**
+ * Test interceptor bindings applied to methods via extension are used to bind the correct interceptor and returned from InvocationContext.getInterceptorBindings()
+ */
+@SpecVersion(spec = "interceptors", version = "2.2")
+public class ChangeInterceptorBindingTest extends AbstractTest {
+    
+    @Deployment
+    public static WebArchive createTestArchive() {
+        return new WebArchiveBuilder()
+                .withTestClassPackage(ChangeInterceptorBindingTest.class)
+                .withBuildCompatibleExtension(ChangeInterceptorBindingExtension.class)
+                .build();
+    }
+    
+    @Test
+    @SpecAssertion(section = Sections.ENHANCEMENT_PHASE, id = "b")
+    @SpecAssertion(section = InterceptorsSections.INVOCATIONCONTEXT, id="o")
+    public void test() {
+        assertEquals(getContextualReference(MyService.class).hello(), "Intercepted(foo): Hello!");
+    }
```

Comment:
I'm not sure if I've used `SpecVersion` and `SpecAssertion` here correctly.

Reply:
That depends on whether you consider this an interceptor specification tests or a CDI specification test.
If it's the former, it should probably instead reside in `/interceptors/tests/...` and then the spec version is correct.
If it's the latter, I'd rather make it a cdi spec with version 4.1.

Reply:
I don't think this qualifies as Interceptors TCK -- it tests that CDI extensions can change the set of interceptor bindings.

Reply:
Changed the spec version to CDI 4.1.